gateway/reverse_proxy.go

```diff
@@ -340,7 +340,7 @@ func (gw *Gateway) TykNewSingleHostReverseProxy(target *url.URL, spec *APISpec,
 		},
 		sp: sync.Pool{
 			New: func() interface{} {
-				buffer := make([]byte, 32*1024)
+				buffer := make([]byte, 2*1024)
```

Comment:
Add a config/env var to configure this buffer size to a reasonable amount? Small api responses require smaller buffers, in different scenarios a larger buffer would make sense. We're sure it doesn't leak, gets improved in latest mux/fork, but it does create allocation pressure so tuning this value is important